### ISSUE TITLE
Update to new Gnome version

### ISSUE THREAD

**Issue by Piotr1215** [FelipeFTN/Emoji-Copy#69]
## Bug Report

**Current Behavior**
Extension is not compatible with the newest version of gnome on Pop!_OS 22.04 `gnome-shell: 42.9-0ubuntu2.1pop0`

```bash
SyntaxError: import declarations may only appear at top level of a module

Stack trace:
  _init@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:23:33
  ExtensionPrefsDialog@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:10:4
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:129:33
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:186:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:22:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Expected behavior/code**
Expected behavior is that the extension works.

**Environment**

- emoji-copy version(s): v 9
- OS: `Linux pop-os 6.9.3-76060903-generic #202405300957~1721174657~22.04~abb7c06 SMP PREEMPT_DYNAMIC Wed J x86_64 x86_64 x86_64 GNU/Linux`

**Possible Solution**

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by FelipeFTN** [comment]
**Hello Buddy! Thanks for reporting this issue! :100:**

Unfortunately, we cannot upgrade older versions of the extension because they run in older versions of Gnome Shell.
So, in order to solve this bug, I highly recommend you to upgrade your Gnome Shell to a newer version! :pray: 

Here are some tips:
1. Check for a System 
- Open "Settings" > "About".
- Click on "Check for Updates."
- If a new version is available, you should see an upgrade prompt. Follow the instructions to upgrade to the latest version of Pop!_OS.

2. Update and Upgrade the System
- Just run: `sudo apt update && sudo apt full-upgrade`

There are some ways to upgrade Gnome Shell without upgrading your system, like **Building from Source** or by adding a `PPA: dev` repository and installing gnome-shell through **apt**! :rocket: 

**I hope this works to you! Let me know if you need some help!**

**Comment by Piotr1215** [comment]
Thank you for a quick response and the pointers. Unfortunately, `Pop!_OS 22.04` comes with pre-built gnome version `42.9` and from what I understand cannot be upgraded independently (due to COSMIC desktop being built on top of that).

I'm not really sure what has changes in the recent system update that affected the emoji extension? Maybe it's not the version but something else?

Thanks anyway, I'll keep digging.

**Comment by Piotr1215** [comment]
Yes, I have tried reinstalling it and also installing earlier version for Gnome 42. 

The extension simply stays `off` and there are no errors in `journalctl | grep emoji-copy`

There must have a change in the recent pop release that disabled it. 

I've also tried other extensions, and they don't work either.

**Comment by FelipeFTN** [comment]
So, looks like it's a problem with **Pop!_OS** using an older version of Gnome, right?

Well, you may try to use some native extension for Emoji Selection, like this one!
> https://github.com/leb-kuchen/cosmic-ext-applet-emoji-selector
![image](https://github.com/user-attachments/assets/8b43bc7f-3486-40de-bafd-3af719b3ed4a)

It seems to work fine, but Just for COSMIC DE... :eyes: 

You got this error by running `make debug`? We still can try something to fix this... But you said every gnome extension in your machine stopped working, right?
> SyntaxError: import declarations may only appear at top level of a module
    Stack trace:
  _init@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:23:33
  ExtensionPrefsDialog@resource:///org/gnome/Shell/Extensions/js/extensionPrefsDialog.js:10:4
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:129:33
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:186:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:22:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17

**Comment by Piotr1215** [comment]
Thank you for pointing me to the `cosmic-ext-applet-emoji-selector`, it looks promising, but still [lacks the shortcut](https://github.com/leb-kuchen/cosmic-ext-applet-emoji-selector/issues/14) for emoji picking, I'm also not sure how to use it :grimacing: 

> But you said every gnome extension in your machine stopped working, right?
Not _all_ extensions, but all the emoji extensions. I found one more, but it also doesn't work (stays _off_)

![image](https://github.com/user-attachments/assets/54fc62bd-c9c0-4441-9bd3-d91813f639de)

**Comment by Piotr1215** [comment]
I'm closing this, since after another update all is back to normal 🤷
